Add `db.stored_procedure.name` to the general span conventions (#2205)
open-telemetry/semantic-conventions: docs/database/database-spans.md

The <span id="target-placeholder">`{target}`</span> SHOULD describe the entity that the operation is performed against
and SHOULD adhere to one of the following values, provided they are accessible:

- `db.collection.name` SHOULD be used for operations on a specific database collection.

> - `db.stored_procedure.name` SHOULD be used for operations that call a specific stored procedure.

Reviewer: would we populate it for DDL? Creating/updating/removing stored procedure? Maybe

```suggestion
- `db.stored_procedure.name` SHOULD be used for operations on a specific stored procedure.
```
